### PR TITLE
Bio.SeqIO support for reading and writing UCSC nib files

### PR DESCRIPTION
**Pull request by mdehoon**
This pull request adds support in Bio.SeqIO for reading and writing DNA sequence files in UCSC's nib file format. This is a binary file format; probably reading and writing could be much faster if this were implemented in C, but Python is good enough for now.

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file and understand that AppVeyor and
TravisCI will be used to confirm the Biopython unit tests and ``flake8`` style
checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

**Comment by MarkusPiotrowski** [comment]
@mdehoon Ever thought about installing flake8 and integrating it into your local git? ;-)

https://flake8.pycqa.org/en/latest/user/using-hooks.html

**Comment by mdehoon** [comment]
The remaining error is surprising - this does not seem related to this pull request, unless I broke test_SeqIO_write.py somehow.

**Comment by peterjc** [comment]
Can you add a test ``*.nib`` file (for which you may need to tell git to tag it as binary), or explicitly check the binary format on disk?

**Comment by peterjc** [comment]
For automating pep8, see #493 for a script which takes into account that we still have per-top-level folder flake8 configurations.

**Comment by peterjc** [comment]
The write test is failing because this format seems not to store any annotation at all, not even a sequence ID or name. You will need to special case the SeqIO write test accordingly.

**Comment by peterjc** [comment]
Need to update the legacy style print-and-compare output:

```
$ python run_tests.py -g test_SeqIO
```

File: Bio/SeqIO/__init__.py [hunk 1]
```diff
@@ -438,6 +439,7 @@
                      "fastq-illumina": QualityIO.FastqIlluminaIterator,
                      "qual": QualityIO.QualPhredIterator,
                      "sff": SffIO.SffIterator,
+                     "nib": NibIO.NibIterator,
```
Comment (peterjc) [inline Bio/SeqIO/__init__.py]: This should be a couple of lines higher up (alphabetical), not in the middle of the two SFF entries.

File: Bio/SeqIO/__init__.py [hunk 2]
```diff
@@ -473,11 +475,12 @@
                    "phd": PhdIO.PhdWriter,
                    "qual": QualityIO.QualPhredWriter,
                    "sff": SffIO.SffWriter,
+                   "nib": NibIO.NibWriter,
```
Comment (peterjc) [inline Bio/SeqIO/__init__.py]: This should be a couple of lines higher up (alphabetical), you could fix the ``pir`` entry too at the same time?

**Comment by peterjc** [comment]
You also need to add ``nib`` to the module docstring listing of supported formats under:

```
File Formats
```

And separately, once this is merged, we'll need to add it to the table on https://github.com/biopython/biopython.github.io/blob/master/wiki/SeqIO.md too for the website.

**Comment by peterjc** [comment]
So just the tests under Python 2 to fix, otherwise looks good 👍

**Pull request merged**